pkg/cmd/openshift-tests/disruption/poll-service/service_controller.go

```diff
@@ -0,0 +1,194 @@
+	"k8s.io/klog/v2"
+)
+
+type PollServiceController struct {
+	nodeName          string
+	clusterIP         string
+	port              string
+	namespaceName     string
+	stopConfigMapName string
+	recorder          monitorapi.RecorderWriter
+	outFile           io.Writer
+
+	configmapLister corelisters.ConfigMapLister
+
+	informersToSync []cache.InformerSynced
+
+	watcherLock sync.Mutex
+	watcher     *watcher
+
+	syncHandler func(ctx context.Context, key string) error
+	queue       workqueue.RateLimitingInterface
+}
+
+type watcher struct {
+	address                 string
+	port                    string
+	newConnectionSampler    *backenddisruption.BackendSampler
+	reusedConnectionSampler *backenddisruption.BackendSampler
+}
+
+func NewPollServiceWatcher(
+	nodeName string,
+	namespaceName string,
+	clusterIP string,
+	port string,
+	recorder monitorapi.RecorderWriter,
+	outFile io.Writer,
+	stopConfigMapName string,
+	configmapInformer coreinformers.ConfigMapInformer,
+) *PollServiceController {
+
+	c := &PollServiceController{
+		nodeName:          nodeName,
+		namespaceName:     namespaceName,
+		clusterIP:         clusterIP,
+		port:              port,
+		recorder:          recorder,
+		stopConfigMapName: stopConfigMapName,
+		outFile:           outFile,
+
+		configmapLister: configmapInformer.Lister(),
+		informersToSync: []cache.InformerSynced{
+			configmapInformer.Informer().HasSynced,
+		},
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ServicePoller"),
+	}
+
+	c.syncHandler = c.syncServicePoller
+
+	configmapInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			c.queue.Add("checK")
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			c.queue.Add("checK")
+		},
+		DeleteFunc: func(obj interface{}) {
+			c.queue.Add("checK")
+		},
+	})
+
+	return c
+}
+
+func (c *PollServiceController) syncServicePoller(ctx context.Context, key string) error {
+	_, err := c.configmapLister.ConfigMaps(c.namespaceName).Get(c.stopConfigMapName)
+	switch {
+	case err == nil:
+		c.removeAllWatchers()
+		return nil
+	case apierrors.IsNotFound(err):
+		// did not find the stopConfigMap
+	case err != nil:
+		return err
+	}
+
+	c.watcherLock.Lock()
+	defer c.watcherLock.Unlock()
+
+	if c.watcher == nil {
+		url := fmt.Sprintf("http://%s", net.JoinHostPort(c.clusterIP, c.port))
+		fmt.Fprintf(c.outFile, "Adding and starting: %v on node/%v\n", url, c.nodeName)
+
+		// the interval locator is unique for every tuple of poller to target, but the backend is per connection type
+		historicalBackendDisruptionDataForNewConnectionsName := fmt.Sprintf("%s-%v-connections", "service", monitorapi.NewConnectionType)
+		historicalBackendDisruptionDataForReusedConnectionsName := fmt.Sprintf("%s-%v-connections", "service", monitorapi.ReusedConnectionType)
+		intervalLocator := fmt.Sprintf("%s-from-node-%v-to-url-%v", "service", c.nodeName, url)
```

Comment:
url will contain slashes likely to mess with parsing.  Perhaps just the IP.  Also, indicate pod or host as the origination.